rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1839,6 +1842,31 @@ public Integer getBtcTransactionConfirmations(Sha256Hash btcTxHash, Sha256Hash b
         return bestChainHeight - block.getHeight() + 1;
     }
 
+    private StoredBlock getBlockKeepingTestnetConsensus(Sha256Hash btcBlockHash) throws BlockStoreException {
+        long rskBlockNumber = 5_148_285;
+
+        boolean networkIsTestnet = bridgeConstants.getBtcParams().equals(NetworkParameters.fromID(NetworkParameters.ID_TESTNET));
+        Sha256Hash blockHash = Sha256Hash.wrap("00000000e8e7b540df01a7067e020fd7e2026bf86289def2283a35120c1af379");
+
+        // DO NOT MODIFY.
+        // This check is needed since this block caused a misbehaviour
+        // for being stored in the cache but not in the storage
+        if (rskExecutionBlock.getNumber() == rskBlockNumber
+            && networkIsTestnet
+            && btcBlockHash.equals(blockHash)
+        ) {
+            byte[] rawBtcBlockHeader = HexUtils.stringHexToByteArray("000000203b5d178405c4e6e7dc07d63d6de5db1342044791721654760c00000000000000796cf6743a036300b43fb3abe6703d04a7999751b6d5744f20327d1175320bd37b954e66ffff001d56dc11ce");
+
+            BtcBlock btcBlockHeader = new BtcBlock(bridgeConstants.getBtcParams(), rawBtcBlockHeader);
+            BigInteger btcBlockChainWork = new BigInteger("000000000000000000000000000000000000000000000ddeb5fbcd969312a77c", 16);
+            int btcBlockNumber = 2_817_125;
+
+            return new StoredBlock(btcBlockHeader, btcBlockChainWork, btcBlockNumber);
+        }
+
+        return btcBlockStore.get(btcBlockHash);
+    }
```

Comment:
```suggestion
    private StoredBlock getBlock(Sha256Hash btcBlockHash) throws BlockStoreException {
        // DO NOT MODIFY.
        // This check is needed since this block caused a misbehaviour in testnet
        // for being stored in the cache but not in the storage
        if (isTestnetSpecialBlock()) {
            return getSpecialBlockKeepingTestnetConsensus();
        }

        return btcBlockStore.get(btcBlockHash);
    }
    
    private boolean isTestnetSpecialBlock() {         
        long rskBlockNumber = 5_148_285;

        boolean networkIsTestnet = bridgeConstants.getBtcParams().equals(NetworkParameters.fromID(NetworkParameters.ID_TESTNET));
        Sha256Hash blockHash = Sha256Hash.wrap("00000000e8e7b540df01a7067e020fd7e2026bf86289def2283a35120c1af379");
        
        return rskExecutionBlock.getNumber() == rskBlockNumber
            && networkIsTestnet
            && btcBlockHash.equals(blockHash)
    }
    
    private StoredBlock getSpecialBlockKeepingTestnetConsensus() {
            byte[] rawBtcBlockHeader = HexUtils.stringHexToByteArray("000000203b5d178405c4e6e7dc07d63d6de5db1342044791721654760c00000000000000796cf6743a036300b43fb3abe6703d04a7999751b6d5744f20327d1175320bd37b954e66ffff001d56dc11ce");

            BtcBlock btcBlockHeader = new BtcBlock(bridgeConstants.getBtcParams(), rawBtcBlockHeader);
            BigInteger btcBlockChainWork = new BigInteger("000000000000000000000000000000000000000000000ddeb5fbcd969312a77c", 16);
            int btcBlockNumber = 2_817_125;
            
            return new StoredBlock(btcBlockHeader, btcBlockChainWork, btcBlockNumber);
    }
```

something like this? wdyt?